src/tainting/Taint_lval_env.ml

```diff
@@ -480,3 +480,21 @@ let to_string taint_to_str
       pending_propagation_dests "[PENDING PROPAGATION DESTS]"
 
 let seq_of_tainted env = LvalMap.to_seq env.tainted
+
+let find_tainted_lvals_of_common_base { tainted; _ } base =
+  let rec eq_base base1 base2 =
+    match (base1, base2) with
+    | IL.Var name1, IL.Var name2 when IL.compare_name name1 name2 = 0 -> true
+    | VarSpecial (This, _), VarSpecial (This, _) -> true
+    | VarSpecial (Super, _), VarSpecial (Super, _) -> true
+    | VarSpecial (Self, _), VarSpecial (Self, _) -> true
+    | VarSpecial (Parent, _), VarSpecial (Parent, _) -> true
+    | ( Mem { e = Fetch { base = base1; rev_offset = [] }; _ },
```

Comment:
Maybe add a comment here that this is pointer dereferencing like `*x`.

Reply:
done.